packages/esm-patient-test-results-app/src/filter/filter-set.tsx

```diff
@@ -22,43 +23,74 @@ const FilterSet: React.FC<FilterSetProps> = ({ hideFilterSetHeader = false }) =>
   const { t } = useTranslation();
   const { resetTree } = useContext(FilterContext);
   const [searchTerm, setSearchTerm] = useState('');
+  const [treeDataFiltered, setTreeDataFiltered] = useState(roots);
   const [showSearchInput, setShowSearchInput] = useState(false);
 
+  useEffect(() => {
+    setTreeDataFiltered(roots);
+  }, [roots]);
+
+  function filterTreeNode(inputValue, treeNode) {
+    // If the tree node's display value contains the user input, or any of its children's display contains the user input, return true
+    if (
+      treeNode?.display.toLowerCase().includes(inputValue.toLowerCase()) ||
+      (treeNode?.subSets && treeNode.subSets.some((child) => filterTreeNode(inputValue, child)))
+    ) {
```

Comment:
```suggestion
    if (
      treeNode && (
        treeNode.display.toLowerCase().includes(inputValue.toLowerCase()) ||
        (treeNode.subSets && treeNode.subSets.some((child) => filterTreeNode(inputValue, child)))
      )
    ) {
```

I'm also a little wary of the recursion here... If the tree structure gets complicated (and the one here might), this can perform quite badly.